TeamShiksha/logoexecutive
# feat - user verification after Sign-Up

## Summary

Implemented email-based user verification system post sign-up.
Addressed Issue #51  

## Description

This pull request user one existing API and introduces one new API endpoints:

####  [Existing] POST API : router.post("/signin", signinController);
1. Upon user creation, an email containing a URL linked with a token generated during sign-up is sent to the user.

####  [New] router.get("/verify", verifyTokenController);
1. This router manages the verification URL from the email. When the URL is clicked, it verifies the existence of the user with the provided token and promptly deletes the token from the user's database instance if found.

#### Additionally:
1. Created sendEmail.js service to load environment variables and send emails using Nodemailer.
2. Implemented a service method in User.js to fetch the user by Token(Generated in the Signup Process).

#### Package Used (Nodemailer)
Nodemailer, a tool in Node.js, helps send emails from the application. It's used here to smoothly handle sending verification emails for users changing their name or email, ensuring secure and reliable communication.

### How to Test the Changes

#### Environment Variables:

Ensure the following environment variables are correctly set in your .env file:
1. BASE_URL: Set this variable to the base URL of your application (e.g., http://localhost:3000).
2. HOST: Set this variable to your SMTP server's host (e.g., smtp.gmail.com).
4. SERVICE: Specify the email service provider (e.g., gmail).
5. EMAIL_PORT: Set the port number for the SMTP server (e.g., 587).
6. SECURE: Set to true for a secure connection or false for a non-secure connection.
7. USER: Placeholder for the sender's email address or username (e.g., example@example.com).
8. PASS: Placeholder for the password or application-specific password associated with the USER account.

#### Configuration Steps for Gmail Credentials:

1. Open your Gmail account and select "Manage your Google Account."
2. Navigate to the "Security" tab in the left panel.
3. Turn on 2-step verification if not enabled.
4. Scroll down to the "App passwords" section and create a new app password.
5. Save this generated password.
6. Ensure that the PASS in the .env file matches the newly created app password, and the USER is the email account for which the app password was created.

#### Testing Steps:
1. Run the application locally and ensure it's functioning properly.
2. Use an API testing tool (e.g., Postman) to trigger the following API endpoints:
3. Send a POST request  "/signin"  with the necessary data (e.g., firstName, lastName, email, password, confirmpassword).
4. Check the email inbox for the received email and click the verification link.
5. Verify that the token is deleted by sending a GET request to "/verify"

### Context (Optional)

This enhancement aims to automate and secure the user verification process, enhancing user experience and system reliability.

### Screenshots or Recordings (Optional)

https://github.com/FrontendArmy/logoexecutive-backend/assets/99191853/40cba857-4fd7-4ef3-a463-479cc7f187bd

## Checklist
